### ISSUE TITLE
An example included in the spec?

### ISSUE THREAD

**Issue by rajido** [BioSchemas/specifications#130]
Developers implementing bioschemas are looking at the specification as well as the examples. Both are useful, but based on feedback from yesterday's workshop few developers mentioned it would help to have at least one example within the spec. For instance we could add a column after the controlled vocabulary column (maybe in json format for complex types) including one example per property. If a property has several expected types, an example could be provided for each type.

**Comment by AlasdairGray** [comment]
We certainly should link to the examples directory within the specification itself. @kcmcleod could you add that.

We had talked about having a complete example at the start of the spec document. Would that work instead of the extra column?

**Comment by ivanmicetic** [comment]
A complete example would be useful for sure but in that case, you would specify only one type of "expected type" instead of all possible cases that could fit in the extra column. E.g. it would be nice to have an example for "identifier" property in all three Expected Types: text, url and PropertyValue.

**Comment by kcmcleod** [comment]
examples are available from the 'link' tab at the top of the profile page.

waiting on feedback to see if that is sufficient.

**Comment by ivanmicetic** [comment]
Links tab points to complete examples but there you have only one implementation per Expected type. The same thing would happen by linking a single example on top of page as @AlasdairGray suggested. But most of the present examples are based on old specifications, have errors and are inconsistent among each other. It would be much useful to have sample examples for all Expected types of each property so that we can use them as (minimal) "gold standards" or "building blocks" approved by bioschemas community.
The consistency among implementations of bioschemas annotations will be possible only with explicit, simple and minimal (meaning expandable if necessary) examples for each property AND expected type combination.

**Comment by ricardoaat** [comment]
Since this is still an ongoing process which means the specifications are constantly improving we need to constantly fix and improve their examples as well. I'm going through all of them in order to have them updated with the current specifications changes. 

It would be of great help if you can point out which of them you found to be out of date, with errors and inconsistent. @ivanmicetic

**Comment by kcmcleod** [comment]
#118 provides further context

**Comment by ivanmicetic** [comment]
Here are some of issues found in examples during Bioschemas workshop in Bologna:
- `license` property in UniProt example is text but specs say license should be `CreativeWork`|`URL`
- `keywords` property in UniProt (and other) uses one text string with comma separated terms, while EGA example uses a list of strings. Specs say CD: ONE. Should the specs be changed to MANY or is it easier to modify all examples?
- `citation` property is missing in UniProt and EGA and other examples, probably due to ambiguous description. What is the difference between citation and publication?
- What about `@id` properties in UniProt example? Are they really needed? What is their meaning?

**Comment by kcmcleod** [comment]
Recommendation from @ljgarcia:

During our meeting on Thu & Fri last week, we found sometimes confusing or time consuming going through full examples. Having one (or two) mini examples per property could help clarify things. Is is possible to add a new column to specifications where these mini examples are added?

Some of the attendees were @ficolo @rajido @ricardoaat @Kerruba Matt and Petr.

**Comment by ricardoaat** [comment]
@ivanmicetic Awesome! we are addressing these issues. Will have them corrected. Thank you!
@kcmcleod An extra column for examples it's a great idea. I've created a pull request which includes a new example column for the profile specifications fields.
Let me know what you think.
https://github.com/BioSchemas/bioschemas.github.io/pull/60

**Comment by ljgarcia** [comment]
Hi @ivanmicetic. Some comments here
- I understand a text corresponding to a URL will be correctly interpreted as url. @danbri (Dan Brickcley) can tell us better, maybe @rajido or @AlasdairGray can ping him?
- keywords. Only one text with keywords separated by commas is actually the schema.org description so we should not use lists here.
- publication is currently used for "how to cite us" while citation is used to cite/reference other CreativeWork. I have suggested to use subjectOf instead publication for "how to cite us", see [issue 128](https://github.com/BioSchemas/specifications/issues/128)
- @id is the node id in json-ld, see [json-ld spec](https://www.w3.org/TR/json-ld/#syntax-tokens-and-keywords), I still have to improve its use in UniProt examples

Hope this helps

**Comment by ljgarcia** [comment]
@ivanmicetic btw, I am not sure what the guidelines in schema.org are regarding "@id", @danbri (Dan Brickley) again is the one that could clarify. @rajido @AlasdairGray maybe you can find out? Using "@id" is a good practice regarding semantic web but maybe it is not that important in schema.org especially for inline elements. for the main defined element, it is indeed important as this is the way others can point to an entity defined somewhere else without repeating/duplicating all the information

**Comment by ivanmicetic** [comment]
@ljgarcia as far as I understood, @id it may be important in bioschemas for example in connecting datasets to a data catalog, if the two entities are described in separate jsons. But it may be redundant if we incorporate both specs in one file (as UniProt and other examples)

**Comment by danbri** [comment]
Regarding ...

 * "I understand a text corresponding to a URL will be correctly interpreted as url."

I don't totally understand the question. Can you give a complete example? In general it is best to be explicit. For the JSON-LD syntax there is the notion of @- context file definitions which say, for each property, whether they default to URLs or strings. This helps with relative URI processing. Currently at Schema.org we decide this based on whether a property has "URL" amongst its expected types.

**Comment by ljgarcia** [comment]
Hi @danbri, thanks for your reply. Here some clarification.

Let's take the property url from Thing, the expected range is the type URL. Is it correct using a string as a URL? I thought yes, but there have been some doubts. 

For example, is this correct?
`
{
  "@type": "Thing",
  "url": "http://anything.url"
}
`

Or should we use URL like this:
`
{
  "@type": "Thing",
  "url":  {
    "@type": "URL",
    ###something else here pointing to the actual url
  }
}
`

**Comment by AlasdairGray** [comment]
I'm going to close this issue as the original focus of this was solved. That is, we have added:
- Links to the example folder 
- Embedded an example within the profile which the GoWeb tool than embeds to the website
- Started embedding examples in the profiles and created issues for those profiles that do not have an example

The discussion around issues with specific examples should be separated out into separate issues if they are still relevant.
